Disable split tunneling for GitLab VPN in prod and anvilprod (DataBiosphere/azul-private#133)

dsotirho-ucsc opened the pull request:
<!--
This is the PR template for regular PRs against `develop`. Edit the URL in your
browser's location bar, appending either `&template=anvilprod-promotion.md`,
`&template=prod-promotion.md`, `&template=anvilprod-hotfix.md`, `&template=prod-
hotfix.md`, `&template=backport.md` or `&template=upgrade.md` to switch the
template.
-->

Connected issues: DataBiosphere/azul-private#133


## Checklist


### Author

- [x] PR is a draft
- [x] Target branch is `develop`
- [x] Name of PR branch matches `issues/<GitHub handle of author>/<issue#>-<slug>`
- [x] On ZenHub, PR is connected to all issues it (partially) resolves
- [x] PR description links to connected issues
- [x] PR title matches<sup>1</sup> that of a connected issue <sub>or comment in PR explains why they're different</sub>
- [x] PR title references all connected issues
- [x] For each connected issue, there is at least one commit whose title references that issue

<sup>1</sup> when the issue title describes a problem, the corresponding PR
title is `Fix: ` followed by the issue title


### Author (partiality)

- [x] Added `p` tag to titles of partial commits
- [x] This PR is labeled `partial` <sub>or completely resolves all connected issues</sub>
- [x] This PR partially resolves each of the connected issues <sub>or does not have the `partial` label</sub>


### Author (chains)

- [x] This PR is blocked by previous PR in the chain <sub>or is not chained to another PR</sub>
- [x] The blocking PR is labeled `base` <sub>or this PR is not chained to another PR</sub>
- [x] This PR is labeled `chained` <sub>or is not chained to another PR</sub>


### Author (reindex, API changes)

- [x] Added `r` tag to commit title <sub>or the changes introduced by this PR will not require reindexing of any deployment</sub>
- [x] This PR is labeled `reindex:dev` <sub>or the changes introduced by it will not require reindexing of `dev`</sub>
- [x] This PR is labeled `reindex:anvildev` <sub>or the changes introduced by it will not require reindexing of `anvildev`</sub>
- [x] This PR is labeled `reindex:anvilprod` <sub>or the changes introduced by it will not require reindexing of `anvilprod`</sub>
- [x] This PR is labeled `reindex:prod` <sub>or the changes introduced by it will not require reindexing of `prod`</sub>
- [x] This PR is labeled `reindex:partial` and its description documents the specific reindexing procedure for `dev`, `anvildev`, `anvilprod` and `prod` <sub>or requires a full reindex or carries none of the labels `reindex:dev`, `reindex:anvildev`, `reindex:anvilprod` and `reindex:prod`</sub>
- [x] This PR and its connected issues are labeled `API` <sub>or this PR does not modify a REST API</sub>
- [x] Added `a` (`A`) tag to commit title for backwards (in)compatible changes <sub>or this PR does not modify a REST API</sub>
- [x] Updated REST API version number in `app.py` <sub>or this PR does not modify a REST API</sub>


### Author (upgrading deployments)

- [x] Ran `make image_manifests.json` and committed the resulting changes <sub>or this PR does not modify `azul_docker_images`, or any other variables referenced in the definition of that variable</sub>
- [x] Documented upgrading of deployments in UPGRADING.rst <sub>or this PR does not require upgrading deployments</sub>
- [x] Added `u` tag to commit title <sub>or this PR does not require upgrading deployments</sub>
- [x] This PR is labeled `upgrade` <sub>or does not require upgrading deployments</sub>
- [x] This PR is labeled `deploy:shared` <sub>or does not modify `image_manifests.json`, and does not require deploying the `shared` component for any other reason</sub>
- [x] This PR is labeled `deploy:gitlab` <sub>or does not require deploying the `gitlab` component</sub>
- [x] This PR is labeled `deploy:runner` <sub>or does not require deploying the `runner` image</sub>


### Author (hotfixes)

- [x] Added `F` tag to main commit title <sub>or this PR does not include permanent fix for a temporary hotfix</sub>
- [x] Reverted the temporary hotfixes for any connected issues <sub>or the none of the stable branches (`anvilprod` and `prod`) have temporary hotfixes for any of the issues connected to this PR</sub>


### Author (before every review)

- [x] Rebased PR branch on `develop`, squashed old fixups
- [x] Ran `make requirements_update` <sub>or this PR does not modify `requirements*.txt`, `common.mk`, `Makefile` and `Dockerfile`</sub>
- [x] Added `R` tag to commit title <sub>or this PR does not modify `requirements*.txt`</sub>
- [x] This PR is labeled `reqs` <sub>or does not modify `requirements*.txt`</sub>
- [x] `make integration_test` passes in personal deployment <sub>or this PR does not modify functionality that could affect the IT outcome</sub>


### Peer reviewer (after approval)

- [x] PR is not a draft
- [x] Ticket is in *Review requested* column
- [x] PR is awaiting requested review from system administrator
- [x] PR is assigned to only the system administrator


### System administrator (after approval)

- [x] Actually approved the PR
- [x] Labeled connected issues as `demo` or `no demo`
- [x] Commented on connected issues about demo expectations <sub>or all connected issues are labeled `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] A comment to this PR details the completed security design review
- [x] PR title is appropriate as title of merge commit
- [x] `N reviews` label is accurate
- [x] Moved ticket to *Approved* column
- [x] PR is assigned to only the operator


### Operator (before pushing merge the commit)

- [x] Checked `reindex:…` labels and `r` commit title tag
- [x] Checked that demo expectations are clear <sub>or all connected issues are labeled `no demo`</sub>
- [x] Squashed PR branch and rebased onto `develop`
- [x] Sanity-checked history
- [x] Pushed PR branch to GitHub
- [x] Ran `_select dev.shared && CI_COMMIT_REF_NAME=develop make -C terraform/shared apply_keep_unused` <sub>or this PR is not labeled `deploy:shared`</sub>
- [x] Ran `_select dev.gitlab && CI_COMMIT_REF_NAME=develop make -C terraform/gitlab apply` <sub>or this PR is not labeled `deploy:gitlab`</sub>
- [x] Ran `_select anvildev.shared && CI_COMMIT_REF_NAME=develop make -C terraform/shared apply_keep_unused` <sub>or this PR is not labeled `deploy:shared`</sub>
- [x] Ran `_select anvildev.gitlab && CI_COMMIT_REF_NAME=develop make -C terraform/gitlab apply` <sub>or this PR is not labeled `deploy:gitlab`</sub>
- [x] Checked the items in the next section <sub>or this PR is labeled `deploy:gitlab`</sub>
- [x] PR is assigned to only the system administrator <sub>or this PR is not labeled `deploy:gitlab`</sub>


### System administrator

- [x] Background migrations for `dev.gitlab` are complete <sub>or this PR is not labeled `deploy:gitlab`</sub>
- [x] Background migrations for `anvildev.gitlab` are complete <sub>or this PR is not labeled `deploy:gitlab`</sub>
- [x] PR is assigned to only the operator


### Operator (before pushing merge the commit)

- [x] Ran `_select dev.gitlab && make -C terraform/gitlab/runner` <sub>or this PR is not labeled `deploy:runner`</sub>
- [x] Ran `_select anvildev.gitlab && make -C terraform/gitlab/runner` <sub>or this PR is not labeled `deploy:runner`</sub>
- [x] Added `sandbox` label <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed PR branch to GitLab `dev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed PR branch to GitLab `anvildev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Reviewed build logs for anomalies in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Reviewed build logs for anomalies in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Deleted unreferenced indices in `sandbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices in `dev`</sub>
- [x] Deleted unreferenced indices in `anvilbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices in `anvildev`</sub>
- [x] Started reindex in `sandbox` <sub>or this PR is not labeled `reindex:dev`</sub>
- [x] Started reindex in `anvilbox` <sub>or this PR is not labeled `reindex:anvildev`</sub>
- [x] Checked for failures in `sandbox` <sub>or this PR is not labeled `reindex:dev`</sub>
- [x] Checked for failures in `anvilbox` <sub>or this PR is not labeled `reindex:anvildev`</sub>
- [x] The title of the merge commit starts with the title of this PR
- [x] Added PR # reference to merge commit title
- [x] Collected commit title tags in merge commit title <sub>but only included `p` if the PR is also labeled `partial`</sub>
- [x] Moved connected issues to Merged column in ZenHub
- [x] Pushed merge commit to GitHub


### Operator (chain shortening)

- [x] Changed the target branch of the blocked PR to `develop` <sub>or this PR is not labeled `base`</sub>
- [x] Removed the `chained` label from the blocked PR <sub>or this PR is not labeled `base`</sub>
- [x] Removed the blocking relationship from the blocked PR <sub>or this PR is not labeled `base`</sub>
- [x] Removed the `base` label from this PR <sub>or this PR is not labeled `base`</sub>


### Operator (after pushing the merge commit)

- [x] Pushed merge commit to GitLab `dev`
- [x] Pushed merge commit to GitLab `anvildev`
- [x] Build passes on GitLab `dev`
- [x] Reviewed build logs for anomalies on GitLab `dev`
- [x] Build passes on GitLab `anvildev`
- [x] Reviewed build logs for anomalies on GitLab `anvildev`
- [x] Ran `_select dev.shared && make -C terraform/shared apply` <sub>or this PR is not labeled `deploy:shared`</sub>
- [x] Ran `_select anvildev.shared && make -C terraform/shared apply` <sub>or this PR is not labeled `deploy:shared`</sub>
- [x] Deleted PR branch from GitHub
- [x] Deleted PR branch from GitLab `dev`
- [x] Deleted PR branch from GitLab `anvildev`


### Operator (reindex)

- [x] Deindexed all unreferenced catalogs in `dev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:dev`</sub>
- [x] Deindexed all unreferenced catalogs in `anvildev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:anvildev`</sub>
- [x] Deindexed specific sources in `dev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:dev`</sub>
- [x] Deindexed specific sources in `anvildev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:anvildev`</sub>
- [x] Indexed specific sources in `dev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:dev`</sub>
- [x] Indexed specific sources in `anvildev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:anvildev`</sub>
- [x] Started reindex in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [x] Started reindex in `anvildev` <sub>or this PR does not require reindexing `anvildev`</sub>
- [x] Checked for, triaged and possibly requeued messages in both fail queues in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [x] Checked for, triaged and possibly requeued messages in both fail queues in `anvildev` <sub>or this PR does not require reindexing `anvildev`</sub>
- [x] Emptied fail queues in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [x] Emptied fail queues in `anvildev` <sub>or this PR does not require reindexing `anvildev`</sub>


### Operator

- [x] Propagated the `deploy:shared`, `deploy:gitlab`, `deploy:runner`, `reindex:partial`, `reindex:anvilprod` and `reindex:prod` labels to the next promotion PRs <sub>or this PR carries none of these labels</sub>
- [x] Propagated any specific instructions related to the `deploy:shared`, `deploy:gitlab`, `deploy:runner`, `reindex:partial`, `reindex:anvilprod` and `reindex:prod` labels, from the description of this PR to that of the next promotion PRs <sub>or this PR carries none of these labels</sub>
- [x] PR is assigned to no one


## Shorthand for review comments

- `L` line is too long
- `W` line wrapping is wrong
- `Q` bad quotes
- `F` other formatting problem

dsotirho-ucsc committed:
[u] Disable split tunneling for GitLab VPN in prod and anvilprod (DataBiosphere/azul-private#133)

achave11-ucsc commented on terraform/gitlab/gitlab.tf.json.template.py:
```diff
@@ -1066,25 +1074,25 @@ def sq(*words):
                 'egress': [
-                    vpc.security_rule(description='Any traffic to the VPC',
-                                      cidr_blocks=['${aws_vpc.gitlab.cidr_block}'],
+                    vpc.security_rule(description=f"Any traffic to the {'VPC' if split_tunnel else 'internet'}",
```
L

achave11-ucsc commented on terraform/gitlab/gitlab.tf.json.template.py:
```diff
@@ -1066,25 +1074,25 @@ def sq(*words):
                 'ingress': [
-                    vpc.security_rule(description='Any traffic from the VPC',
-                                      cidr_blocks=['${aws_vpc.gitlab.cidr_block}'],
+                    vpc.security_rule(description=f"Any traffic from the {'VPC' if split_tunnel else 'internet'}",
```
L

achave11-ucsc commented on terraform/gitlab/gitlab.tf.json.template.py:
```diff
@@ -1226,10 +1239,19 @@ def sq(*words):
+        'aws_ec2_client_vpn_route': {
+            f'gitlab_{zone}': {
+                'client_vpn_endpoint_id': '${aws_ec2_client_vpn_endpoint.gitlab.id}',
+                'target_vpc_subnet_id': f'${{aws_subnet.gitlab_public_{zone}.id}}',
```
We prefer the following notation when interpolating strings in combination of the Terraform syntax,
```suggestion
                'target_vpc_subnet_id': '${aws_subnet.gitlab_public_%s.id}' % zone,
```

achave11-ucsc reviewed (changes requested):
Also, IT file and such additional checks seem to be missing.

dsotirho-ucsc commented:
[6099_IT_2024-04-23.txt](https://github.com/DataBiosphere/azul/files/15101540/6099_IT_2024-04-23.txt)

achave11-ucsc reviewed (dismissed):
LGTM

hannes-ucsc commented on UPGRADING.rst:
```diff
@@ -20,6 +20,14 @@ reverted. This is all fairly informal and loosely defined. Hopefully we won't
+
+Remove the line: `pull-filter ignore redirect-gateway` from your `anvilprod` VPN
+connection. In Viscosity, the configuration can be found at: Settings -> (Select
+connection) -> Edit -> Advanced
```
```suggestion
Remove the line: `pull-filter ignore redirect-gateway` from your `anvilprod` and `prod` VPN
connections. In Viscosity, the configuration can be found at: Settings -> (Select
connection) -> Edit -> Advanced
```

Noa doesn't use Viscosity. It would be better to have people run the `make config` target and reimport the connection.

hannes-ucsc committed:
Convert config.is_stable_deployment to property

hannes-ucsc committed:
Fix: Diagram shows `anvilprod` as lower deployment (#6147)

hannes-ucsc committed:
Derive unstable deployment status from branch names (#6147)

hannes-ucsc reviewed (changes requested):
Please incorporate the changes from my branch [issues/hannes-ucsc/133-stable](https://github.com/DataBiosphere/azul/tree/issues/hannes-ucsc/133-stable) as early commits in your branch.

dsotirho-ucsc commented:
[6192_IT_2024-04-25.txt](https://github.com/DataBiosphere/azul/files/15123546/6192_IT_2024-04-25.txt)

hannes-ucsc commented on terraform/gitlab/vpn/Makefile:
```diff
@@ -57,7 +57,9 @@ config: _client
 			--client-vpn-endpoint-id $$endpoint \
 			--output=text
-	@echo "pull-filter ignore redirect-gateway"
+	@if test "$$(AZUL_DEBUG=0 python -m azul 'config.is_stable_deployment')" != "True"; then \
```
```suggestion
	@if [ "$$(AZUL_DEBUG=0 python -m azul config.is_stable_deployment)" != True ]; then \
```

hannes-ucsc replied:
To be consistent with line 51. Only quote what needs to be quoted.

hannes-ucsc commented on UPGRADING.rst:
```diff
@@ -20,6 +20,19 @@ reverted. This is all fairly informal and loosely defined. Hopefully we won't
+
+    _select prod.gitlab  # or anvilprod.gitlab
+    cd terraform/gitlab/vpn
+    git submodule update --init easy-rsa
```
```suggestion
```

At this point everyone must have the submodule updated.

I recommend using `git config submodule.recurse true` once in the source tree so that submodules are updated automatically on `git pull`.

hannes-ucsc replied:
Also ask to delete the ovpn files after import so that the private key doesn't proliferate

hannes-ucsc commented on UPGRADING.rst:
```diff
@@ -20,6 +20,19 @@ reverted. This is all fairly informal and loosely defined. Hopefully we won't
+DataBiosphere/azul-private#133 Disable split tunneling for GitLab VPN in prod and anvilprod
+===========================================================================================
+
+This change requires an update to your existing VPN connections for `prod` and
```
Confusing ordering of actions in the second sentence. This should look like

> This change requires …
> 
> Run the following commands:
> 
> ```
> <commands>
> ```
>
> Then …

dsotirho-ucsc commented:
[6192_IT_2024-04-26.txt](https://github.com/DataBiosphere/azul/files/15134255/6192_IT_2024-04-26.txt)

hannes-ucsc commented:
### Security design review

- [x] Security design review completed; this PR does **not** …
  - [x] … affect authentication; for example:
    - OAuth 2.0 with the application (API or Swagger UI)
    - Authentication of developers with Google Cloud APIs
    - Authentication of developers with AWS APIs
    - Authentication with a GitLab instance in the system
    - Password and 2FA authentication with GitHub
    - API access token authentication with GitHub
    - Authentication with Terra 
  - [x] … affect the permissions of internal users like access to
    - Cloud resources on AWS and GCP
    - GitLab repositories, projects and groups, administration
    - an EC2 instance via SSH
    - GitHub issues, pull requests, commits, commit statuses, wikis, repositories, organizations
  - [x] … affect the permissions of external users like access to
    - TDR snapshots
  - [x] … affect permissions of service or bot accounts
    - Cloud resources on AWS and GCP
  - [x] … affect audit logging in the system, like
    - adding, removing or changing a log message that represents an auditable event
    - changing the routing of log messages through the system
  - [x] … affect monitoring of the system
  - [x] … introduce a new software dependency like
    - Python packages on PYPI
    - Command-line utilities
    - Docker images
    - Terraform providers
  - [x] … add an interface that exposes sensitive or confidential data at the security boundary
  - [x] … affect the encryption of data at rest
  - [x] … require persistence of sensitive or confidential data that might require encryption at rest
  - [x] … require unencrypted transmission of data within the security boundary
  - [x] … affect the network security layer; for example by 
    - modifying, adding or removing firewall rules
    - modifying, adding or removing security groups
    - changing or adding a port a service, proxy or load balancer listens on
- [x] Documentation on any unchecked boxes is provided in comments below

hannes-ucsc commented:
Security design review: This change disables split tunneling on stable aka "production" deployments. While not necessarily improving our security posture (that's the reason I argued against implementing it) it doesn't worsen it either.

dsotirho-ucsc commented:
dev:

```
Terraform will perform the following actions:

  # aws_security_group.gitlab_alb will be updated in-place
  ~ resource "aws_security_group" "gitlab_alb" {
      ~ egress                 = [
          - {
              - cidr_blocks      = [
                  - "0.0.0.0/0",
                ]
              - description      = "ICMP for PMTUD"
              - from_port        = 3
              - ipv6_cidr_blocks = []
              - prefix_list_ids  = []
              - protocol         = "icmp"
              - security_groups  = []
              - self             = false
              - to_port          = 4
            },
          + {
              + cidr_blocks      = []
              + description      = "ICMP for PMTUD"
              + from_port        = 3
              + ipv6_cidr_blocks = []
              + prefix_list_ids  = []
              + protocol         = "icmp"
              + security_groups  = []
              + self             = false
              + to_port          = 4
            },
            # (1 unchanged element hidden)
        ]
        id                     = "sg-0919cfd4671d09935"
        name                   = "azul-gitlab-alb"
        tags                   = {
            "Name"                = "azul-gitlab_alb"
            "billing"             = "hca"
            "component"           = "azul-gitlab_alb"
            "deployment"          = "dev"
            "owner"               = "hannes@ucsc.edu"
            "service"             = "azul"
            "terraform_component" = "gitlab"
        }
        # (7 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

prod:
```
Terraform will perform the following actions:

  # aws_ec2_client_vpn_authorization_rule.gitlab must be replaced
-/+ resource "aws_ec2_client_vpn_authorization_rule" "gitlab" {
      ~ id                     = "cvpn-endpoint-05749d371fb1e4340,172.22.0.0/16" -> (known after apply)
      ~ target_network_cidr    = "172.22.0.0/16" -> "0.0.0.0/0" # forces replacement
        # (2 unchanged attributes hidden)
    }

  # aws_ec2_client_vpn_endpoint.gitlab will be updated in-place
  ~ resource "aws_ec2_client_vpn_endpoint" "gitlab" {
      ~ dns_servers            = [
          + "172.22.0.2",
          + "169.254.169.253",
        ]
        id                     = "cvpn-endpoint-05749d371fb1e4340"
      ~ split_tunnel           = true -> false
        tags                   = {
            "Name"                = "azul-gitlab"
            "billing"             = "hca"
            "component"           = "azul-gitlab"
            "deployment"          = "prod"
            "owner"               = "hannes@ucsc.edu"
            "service"             = "azul"
            "terraform_component" = "gitlab"
        }
        # (11 unchanged attributes hidden)

        # (4 unchanged blocks hidden)
    }

  # aws_ec2_client_vpn_route.gitlab_0 will be created
  + resource "aws_ec2_client_vpn_route" "gitlab_0" {
      + client_vpn_endpoint_id = "cvpn-endpoint-05749d371fb1e4340"
      + destination_cidr_block = "0.0.0.0/0"
      + id                     = (known after apply)
      + origin                 = (known after apply)
      + target_vpc_subnet_id   = "subnet-0684c75a115306ba9"
      + type                   = (known after apply)
    }

  # aws_ec2_client_vpn_route.gitlab_1 will be created
  + resource "aws_ec2_client_vpn_route" "gitlab_1" {
      + client_vpn_endpoint_id = "cvpn-endpoint-05749d371fb1e4340"
      + destination_cidr_block = "0.0.0.0/0"
      + id                     = (known after apply)
      + origin                 = (known after apply)
      + target_vpc_subnet_id   = "subnet-086a7d085dffe194f"
      + type                   = (known after apply)
    }

  # aws_instance.gitlab must be replaced
-/+ resource "aws_instance" "gitlab" {
      ~ arn                                  = "arn:aws:ec2:us-east-1:542754589326:instance/i-0768094e4bd0f21d5" -> (known after apply)
      ~ associate_public_ip_address          = false -> (known after apply)
      ~ availability_zone                    = "us-east-1a" -> (known after apply)
      ~ cpu_core_count                       = 2 -> (known after apply)
      ~ cpu_threads_per_core                 = 2 -> (known after apply)
      ~ disable_api_stop                     = false -> (known after apply)
      ~ disable_api_termination              = false -> (known after apply)
      ~ ebs_optimized                        = false -> (known after apply)
      - hibernation                          = false -> null
      + host_id                              = (known after apply)
      + host_resource_group_arn              = (known after apply)
      ~ id                                   = "i-0768094e4bd0f21d5" -> (known after apply)
      ~ instance_initiated_shutdown_behavior = "stop" -> (known after apply)
      + instance_lifecycle                   = (known after apply)
      ~ instance_state                       = "running" -> (known after apply)
      ~ ipv6_address_count                   = 0 -> (known after apply)
      ~ ipv6_addresses                       = [] -> (known after apply)
      ~ monitoring                           = false -> (known after apply)
      + outpost_arn                          = (known after apply)
      + password_data                        = (known after apply)
      + placement_group                      = (known after apply)
      ~ placement_partition_number           = 0 -> (known after apply)
      ~ primary_network_interface_id         = "eni-0fad7f478537f0190" -> (known after apply)
      ~ private_dns                          = "ip-172-22-0-54.ec2.internal" -> (known after apply)
      ~ private_ip                           = "172.22.0.54" -> (known after apply)
      + public_dns                           = (known after apply)
      + public_ip                            = (known after apply)
      ~ secondary_private_ips                = [] -> (known after apply)
      ~ security_groups                      = [] -> (known after apply)
      - source_dest_check                    = true -> null
      + spot_instance_request_id             = (known after apply)
      ~ subnet_id                            = "subnet-08a76b22ec37e1a07" -> (known after apply)
        tags                                 = {
            "Name"                = "azul-gitlab"
            "Owner"               = "hannes@ucsc.edu"
            "billing"             = "hca"
            "component"           = "azul-gitlab"
            "deployment"          = "prod"
            "owner"               = "hannes@ucsc.edu"
            "service"             = "azul"
            "terraform_component" = "gitlab"
        }
      ~ tenancy                              = "default" -> (known after apply)
      ~ user_data                            = "28a41eadfa38581a571c38e1e51baf4a2afb4708" -> "19b3e22d1bba17d4b5d865207af628e7e0876f9f" # forces replacement
      + user_data_base64                     = (known after apply)
      ~ vpc_security_group_ids               = [
          - "sg-0c954a5a664444fe4",
        ] -> (known after apply)
        # (7 unchanged attributes hidden)

      - capacity_reservation_specification {
          - capacity_reservation_preference = "open" -> null
        }

      - cpu_options {
          - core_count       = 2 -> null
          - threads_per_core = 2 -> null
        }

      - credit_specification {
          - cpu_credits = "unlimited" -> null
        }

      - ebs_block_device {
          - delete_on_termination = false -> null
          - device_name           = "/dev/sdf" -> null
          - encrypted             = true -> null
          - iops                  = 300 -> null
          - kms_key_id            = "arn:aws:kms:us-east-1:542754589326:key/606fb0cb-81df-4a09-967c-0c23853155f6" -> null
          - snapshot_id           = "snap-08c97e317c09f3e57" -> null
          - tags                  = {
              - "Name"  = "azul-gitlab"
              - "owner" = "hannes@ucsc.edu"
            } -> null
          - throughput            = 0 -> null
          - volume_id             = "vol-07a1832cc7b7273b8" -> null
          - volume_size           = 100 -> null
          - volume_type           = "gp2" -> null
        }

      - enclave_options {
          - enabled = false -> null
        }

      - maintenance_options {
          - auto_recovery = "default" -> null
        }

      ~ metadata_options {
          ~ instance_metadata_tags      = "disabled" -> (known after apply)
            # (3 unchanged attributes hidden)
        }

      - private_dns_name_options {
          - enable_resource_name_dns_a_record    = false -> null
          - enable_resource_name_dns_aaaa_record = false -> null
          - hostname_type                        = "ip-name" -> null
        }

      ~ root_block_device {
          ~ device_name           = "/dev/xvda" -> (known after apply)
          ~ iops                  = 100 -> (known after apply)
          ~ kms_key_id            = "arn:aws:kms:us-east-1:542754589326:key/606fb0cb-81df-4a09-967c-0c23853155f6" -> (known after apply)
          - tags                  = {} -> null
          ~ throughput            = 0 -> (known after apply)
          ~ volume_id             = "vol-014d31ca7ff4a052b" -> (known after apply)
          ~ volume_type           = "gp2" -> (known after apply)
            # (3 unchanged attributes hidden)
        }

        # (1 unchanged block hidden)
    }

  # aws_lb_target_group_attachment.gitlab_git must be replaced
-/+ resource "aws_lb_target_group_attachment" "gitlab_git" {
      ~ id               = "arn:aws:elasticloadbalancing:us-east-1:542754589326:targetgroup/azul-gitlab-git/75871b91a714874c-20240426213940893500000003" -> (known after apply)
      ~ target_id        = "i-0768094e4bd0f21d5" # forces replacement -> (known after apply) # forces replacement
        # (1 unchanged attribute hidden)
    }

  # aws_lb_target_group_attachment.gitlab_http must be replaced
-/+ resource "aws_lb_target_group_attachment" "gitlab_http" {
      ~ id               = "arn:aws:elasticloadbalancing:us-east-1:542754589326:targetgroup/azul-gitlab-http/630db89565bf1f3d-20240426213940794800000002" -> (known after apply)
      ~ target_id        = "i-0768094e4bd0f21d5" # forces replacement -> (known after apply) # forces replacement
        # (1 unchanged attribute hidden)
    }

  # aws_lb_target_group_attachment.gitlab_ssh must be replaced
-/+ resource "aws_lb_target_group_attachment" "gitlab_ssh" {
      ~ id               = "arn:aws:elasticloadbalancing:us-east-1:542754589326:targetgroup/azul-gitlab-ssh/3f5ad93fceb45135-20240426213941098600000004" -> (known after apply)
      ~ target_id        = "i-0768094e4bd0f21d5" # forces replacement -> (known after apply) # forces replacement
        # (1 unchanged attribute hidden)
    }

  # aws_security_group.gitlab_alb will be updated in-place
  ~ resource "aws_security_group" "gitlab_alb" {
      ~ egress                 = [
          - {
              - cidr_blocks      = [
                  - "0.0.0.0/0",
                ]
              - description      = "ICMP for PMTUD"
              - from_port        = 3
              - ipv6_cidr_blocks = []
              - prefix_list_ids  = []
              - protocol         = "icmp"
              - security_groups  = []
              - self             = false
              - to_port          = 4
            },
          + {
              + cidr_blocks      = []
              + description      = "ICMP for PMTUD"
              + from_port        = 3
              + ipv6_cidr_blocks = []
              + prefix_list_ids  = []
              + protocol         = "icmp"
              + security_groups  = []
              + self             = false
              + to_port          = 4
            },
            # (1 unchanged element hidden)
        ]
        id                     = "sg-08a51da6f2fbbf164"
        name                   = "azul-gitlab-alb"
        tags                   = {
            "Name"                = "azul-gitlab_alb"
            "billing"             = "hca"
            "component"           = "azul-gitlab_alb"
            "deployment"          = "prod"
            "owner"               = "hannes@ucsc.edu"
            "service"             = "azul"
            "terraform_component" = "gitlab"
        }
        # (7 unchanged attributes hidden)
    }

  # aws_security_group.gitlab_vpn will be updated in-place
  ~ resource "aws_security_group" "gitlab_vpn" {
      ~ egress                 = [
          - {
              - cidr_blocks      = [
                  - "172.22.0.0/16",
                ]
              - description      = "Any traffic to the VPC"
              - from_port        = 0
              - ipv6_cidr_blocks = []
              - prefix_list_ids  = []
              - protocol         = "-1"
              - security_groups  = []
              - self             = false
              - to_port          = 0
            },
          + {
              + cidr_blocks      = [
                  + "0.0.0.0/0",
                ]
              + description      = "Any traffic to the internet"
              + from_port        = 0
              + ipv6_cidr_blocks = []
              + prefix_list_ids  = []
              + protocol         = "-1"
              + security_groups  = []
              + self             = false
              + to_port          = 0
            },
            # (1 unchanged element hidden)
        ]
        id                     = "sg-0aa1f5ab5342fe890"
      ~ ingress                = [
          - {
              - cidr_blocks      = [
                  - "172.22.0.0/16",
                ]
              - description      = "Any traffic from the VPC"
              - from_port        = 0
              - ipv6_cidr_blocks = []
              - prefix_list_ids  = []
              - protocol         = "-1"
              - security_groups  = []
              - self             = false
              - to_port          = 0
            },
          + {
              + cidr_blocks      = [
                  + "0.0.0.0/0",
                ]
              + description      = "Any traffic from the internet"
              + from_port        = 0
              + ipv6_cidr_blocks = []
              + prefix_list_ids  = []
              + protocol         = "-1"
              + security_groups  = []
              + self             = false
              + to_port          = 0
            },
            # (1 unchanged element hidden)
        ]
        name                   = "azul-gitlab-vpn"
        tags                   = {
            "Name"                = "azul-gitlab_vpn"
            "billing"             = "hca"
            "component"           = "azul-gitlab_vpn"
            "deployment"          = "prod"
            "owner"               = "hannes@ucsc.edu"
            "service"             = "azul"
            "terraform_component" = "gitlab"
        }
        # (6 unchanged attributes hidden)
    }

  # aws_volume_attachment.gitlab must be replaced
-/+ resource "aws_volume_attachment" "gitlab" {
      ~ id          = "vai-2623030600" -> (known after apply)
      ~ instance_id = "i-0768094e4bd0f21d5" # forces replacement -> (known after apply) # forces replacement
        # (2 unchanged attributes hidden)
    }

Plan: 8 to add, 3 to change, 6 to destroy.
```

hannes-ucsc commented:
Why do have to review this again?

dsotirho-ucsc commented:
> Why do have to review this again?

There was a bug that required a fixup commit.

hannes-ucsc commented:
It appears that deployment of `gitlab.dev` and `gitlab.anvildev` was skipped even though this PR changes the `gitlab` TF component code and the CL doesn't allow for skipping. Please deploy `gitlab.dev` and `gitlab.anvildev` now from `develop`.

dsotirho-ucsc commented:
Deployment of `gitlab.dev` and `gitlab.anvildev` were performed as outlined in the checklists.

```
(.venv) daniel@Crispin ~/repo/azul3 $ git co issues/dsotirho-ucsc/133-disable-split-tunneling
Switched to branch 'issues/dsotirho-ucsc/133-disable-split-tunneling'
Your branch is up to date with 'github/issues/dsotirho-ucsc/133-disable-split-tunneling'.

(.venv) daniel@Crispin ~/repo/azul3 $ git prv
HEAD is now at edcfef0cf fixup! [u] Disable split tunneling for GitLab VPN in prod and anvilprod (DataBiosphere/azul-private#133)

(.venv) daniel@Crispin ~/repo/azul3 $ git log
commit edcfef0cf6ca50fb21808bf6040a8912170db961 (HEAD -> issues/dsotirho-ucsc/133-disable-split-tunneling, github/issues/dsotirho-ucsc/133-disable-split-tunneling)
Author: Daniel Sotirhos <dsotirho@ucsc.edu>
Date:   Mon Apr 29 16:43:05 2024 -0700

    fixup! [u] Disable split tunneling for GitLab VPN in prod and anvilprod (DataBiosphere/azul-private#133)

…

(.venv) daniel@Crispin ~/repo/azul3 $ git rebase -i develop
[detached HEAD 5dcfc3df3] [u] Disable split tunneling for GitLab VPN in prod and anvilprod (DataBiosphere/azul-private#133)
 Date: Mon Apr 22 16:09:25 2024 -0700
 3 files changed, 71 insertions(+), 16 deletions(-)
Successfully rebased and updated refs/heads/issues/dsotirho-ucsc/133-disable-split-tunneling.

(.venv) daniel@Crispin ~/repo/azul3 $ git log
commit 5dcfc3df3feea4cbcdcd7d182942b650d9f24b2a (HEAD -> issues/dsotirho-ucsc/133-disable-split-tunneling)
Author: Daniel Sotirhos <dsotirho@ucsc.edu>
Date:   Mon Apr 22 16:09:25 2024 -0700

    [u] Disable split tunneling for GitLab VPN in prod and anvilprod (DataBiosphere/azul-private#133)

…

(.venv) daniel@Crispin ~/repo/azul3 $ git pfwl
Enumerating objects: 36, done.
Counting objects: 100% (36/36), done.
Delta compression using up to 12 threads
Compressing objects: 100% (17/17), done.
Writing objects: 100% (23/23), 3.73 KiB | 3.73 MiB/s, done.
Total 23 (delta 16), reused 8 (delta 6), pack-reused 0
remote: Resolving deltas: 100% (16/16), completed with 11 local objects.
To github.com:DataBiosphere/azul.git
 + edcfef0cf...5dcfc3df3 issues/dsotirho-ucsc/133-disable-split-tunneling -> issues/dsotirho-ucsc/133-disable-split-tunneling (forced update)


(.venv) daniel@Crispin ~/repo/azul3 $
(.venv) daniel@Crispin ~/repo/azul3 $ _select dev.gitlab
remove .active? y
envhook.py: Currently disabled because the ENVHOOK environment variable is set to 0.
export_environment.py: Loading environment from deployments/dev.gitlab/environment.local.py

…

Quota project "platform-hca-dev" was added to ADC which can be used by Google client libraries for billing and quota. Note that some services may still bill the project owning the resource.
Enter MFA code for arn:aws:iam::652235167018:mfa/dsotirho@ucsc.edu:
Login Succeeded
Login Succeeded
Session credentials are in effect for AWS. Additionally, you have been logged into Google Cloud, Amazon ECR and the GitLab Docker registry. Use '_logout' to invalidate the session credentials. Alternatively, you can use _logout_completely to invalidate all credentials but this is usually not necessary.

(.venv) daniel@Crispin ~/repo/azul3 $
(.venv) daniel@Crispin ~/repo/azul3 $ CI_COMMIT_REF_NAME=develop make -C terraform/gitlab apply
python /Users/daniel/repo/azul3/scripts/check_branch.py
python providers.tf.json.template.py providers.tf.json

…

Terraform will perform the following actions:

  # aws_security_group.gitlab_alb will be updated in-place
  ~ resource "aws_security_group" "gitlab_alb" {
      ~ egress                 = [
          - {
              - cidr_blocks      = [
                  - "0.0.0.0/0",
                ]
              - description      = "ICMP for PMTUD"
              - from_port        = 3
              - ipv6_cidr_blocks = []
              - prefix_list_ids  = []
              - protocol         = "icmp"
              - security_groups  = []
              - self             = false
              - to_port          = 4
            },
          + {
              + cidr_blocks      = []
              + description      = "ICMP for PMTUD"
              + from_port        = 3
              + ipv6_cidr_blocks = []
              + prefix_list_ids  = []
              + protocol         = "icmp"
              + security_groups  = []
              + self             = false
              + to_port          = 4
            },
            # (1 unchanged element hidden)
        ]
        id                     = "sg-0919cfd4671d09935"
        name                   = "azul-gitlab-alb"
        tags                   = {
            "Name"                = "azul-gitlab_alb"
            "billing"             = "hca"
            "component"           = "azul-gitlab_alb"
            "deployment"          = "dev"
            "owner"               = "hannes@ucsc.edu"
            "service"             = "azul"
            "terraform_component" = "gitlab"
        }
        # (7 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_security_group.gitlab_alb: Modifying... [id=sg-0919cfd4671d09935]
aws_security_group.gitlab_alb: Modifications complete after 1s [id=sg-0919cfd4671d09935]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.


(.venv) daniel@Crispin ~/repo/azul3 $
(.venv) daniel@Crispin ~/repo/azul3 $ _select anvildev.gitlab
remove .active? y
envhook.py: Currently disabled because the ENVHOOK environment variable is set to 0.


…

Quota project "platform-anvil-dev" was added to ADC which can be used by Google client libraries for billing and quota. Note that some services may still bill the project owning the resource.
Enter MFA code for arn:aws:iam::652235167018:mfa/dsotirho@ucsc.edu:
Login Succeeded
Login Succeeded
Session credentials are in effect for AWS. Additionally, you have been logged into Google Cloud, Amazon ECR and the GitLab Docker registry. Use '_logout' to invalidate the session credentials. Alternatively, you can use _logout_completely to invalidate all credentials but this is usually not necessary.
(.venv) daniel@Crispin ~/repo/azul3 $

(.venv) daniel@Crispin ~/repo/azul3 $
(.venv) daniel@Crispin ~/repo/azul3 $ CI_COMMIT_REF_NAME=develop make -C terraform/gitlab apply
python /Users/daniel/repo/azul3/scripts/check_branch.py
python providers.tf.json.template.py providers.tf.json
Creating providers.tf.json

…

Terraform will perform the following actions:

  # aws_security_group.gitlab_alb will be updated in-place
  ~ resource "aws_security_group" "gitlab_alb" {
      ~ egress                 = [
          - {
              - cidr_blocks      = [
                  - "0.0.0.0/0",
                ]
              - description      = "ICMP for PMTUD"
              - from_port        = 3
              - ipv6_cidr_blocks = []
              - prefix_list_ids  = []
              - protocol         = "icmp"
              - security_groups  = []
              - self             = false
              - to_port          = 4
            },
          + {
              + cidr_blocks      = []
              + description      = "ICMP for PMTUD"
              + from_port        = 3
              + ipv6_cidr_blocks = []
              + prefix_list_ids  = []
              + protocol         = "icmp"
              + security_groups  = []
              + self             = false
              + to_port          = 4
            },
            # (1 unchanged element hidden)
        ]
        id                     = "sg-0dce6ad44ae1590d5"
        name                   = "azul-gitlab-alb"
        tags                   = {
            "Name"                = "azul-gitlab_alb"
            "billing"             = "anvil"
            "component"           = "azul-gitlab_alb"
            "deployment"          = "anvildev"
            "owner"               = "hannes@ucsc.edu"
            "service"             = "azul"
            "terraform_component" = "gitlab"
        }
        # (7 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_security_group.gitlab_alb: Modifying... [id=sg-0dce6ad44ae1590d5]
aws_security_group.gitlab_alb: Modifications complete after 1s [id=sg-0dce6ad44ae1590d5]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

(.venv) daniel@Crispin ~/repo/azul3 $
(.venv) daniel@Crispin ~/repo/azul3 $ git b
  develop
  issues/achave11-ucsc/5958-manifest-tagging-race
* issues/dsotirho-ucsc/133-disable-split-tunneling
  issues/dsotirho-ucsc/5555-rate-limit-on-http-method
  issues/dsotirho-ucsc/6-content-security-policy
  issues/dsotirho-ucsc/6134-api-unauthorized-awsconfig
  issues/nadove-ucsc/5999-filter-accessible
  prod

(.venv) daniel@Crispin ~/repo/azul3 $ git push gitlab.hcadev ; git push gitlab.anvildev
Enumerating objects: 36, done.
Counting objects: 100% (36/36), done.

…
```

hannes-ucsc commented:
https://ucsc-gi.slack.com/archives/C705Y6G9Z/p1714578472405329
